webapp/src/components/systemConsole/questions/questions.tsx

```diff
@@ -0,0 +1,147 @@
+    id: string;
+    title?: string;
+    text?: string;
+    type: QuestionType;
+    system: boolean;
+    mandatory: boolean;
+    helpText?: string;
+};
+
+function SurveyQuestions({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) {
+    const [questions, setQuestions] = useState<Question[]>([]);
+
+    const generateDefaultQuestions = (): Question[] => {
+        return [
+            {
+                id: utils.uuid(),
+                title: 'Survey message text',
+                text: 'Please take a few moments to help us improve your experience with Mattermost.',
+                type: 'text',
+                system: false,
+                mandatory: true,
+                helpText: 'This text will be sent in the bot message preceding the survey.',
+            },
+            {
+                id: utils.uuid(),
+                text: 'How likely are you to recommend Mattermost?',
+                type: 'linear_scale',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                text: 'How can we make your experience better?',
+                type: 'text',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                type: 'text',
+                system: false,
+                mandatory: false,
+            },
+        ];
+    };
+
+    useEffect(() => {
+        // Set initial value from saved config
+        const savedQuestions = config.PluginSettings.Plugins?.['com.mattermost.user-survey']?.systemconsolesetting.SurveyQuestions;
+
+        const initialSetting = savedQuestions || generateDefaultQuestions();
+        setQuestions(initialSetting);
+        setInitialSetting(id, initialSetting);
+    }, [config.PluginSettings.Plugins, id, setInitialSetting]);
+
+    const saveSettings = useCallback((setting: SurveyQuestionsConfig) => {
+        setSaveNeeded();
+        onChange(id, setting);
+    }, [id, onChange, setSaveNeeded]);
+
+    const questionOnChangeHandler = useDebouncedCallback(
+        useCallback((e: React.ChangeEvent<HTMLInputElement>, questionID: string) => {
+            const newQuestions = [...questions];
+            const i = questions.findIndex((q) => q.id === questionID);
+            newQuestions[i] = {
+                ...newQuestions[i],
+                text: e.target.value,
+            };
+            setQuestions(newQuestions);
+            saveSettings(newQuestions);
+        }, [questions, saveSettings]),
+        500,
+    );
+
+    const renderQuestions = useMemo(() => {
+        return questions.map((question) => {
+            return (
+                <div
+                    key={question.id}
+                    className='vertical question'
+                >
+                    <span className='questionTitle'>
+                        {question.title && question.title}
```

Comment:
I see what you are doing here, but it is really confusing. Why not moving this logic outside of the render like:

```
title = question.title ? question.title : (`${questionTypeDisplayName.get(question.type)} ${question.mandatory ? '' : '(Optional)'}`);
return (
    <div ...>
        <span className='questionTitle'>
            {title}
        </span>
        ...
   </div>
);
```

Maybe making it a bit different so it is better readable.

Reply:
Yeah not sure why we're `&&`ing this. If it's blank/null it just won't get rendered right?

Reply:
Okay, so the part about `question.title` and `question.helpText` are removed now as they were used by the survey message text, which is not a question. I misinterpreted it as a question and added it in the array of questions. So took it out of questions array and thus removed title and helpText fields from question type. So this part of code is gone now.